modules/renter/hostdb/hostdb.go

```diff
@@ -107,11 +108,13 @@ func newHostDB(cs consensusSet, d dialer, s sleeper, p persister, l *persist.Log
 		log:     l,
 
 		// TODO: should index by pubkey, not ip
-		activeHosts: make(map[modules.NetAddress]*hostNode),
+		activeHosts: make(map[modules.NetAddress]*hostEntry),
 		allHosts:    make(map[modules.NetAddress]*hostEntry),
 		scanPool:    make(chan *hostEntry, scanPoolSize),
 	}
 
+	hdb.hostTree = hosttree.New(hdb.calculateHostWeight())
```

Comment:
The reason for this is that the host weight function needs access to the hostdb's state.